### PR TITLE
compute and store the limit violations after the loadflow

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

**Review by antoinebhs — APPROVED** [review]
Code OK
Tests OK

File: src/main/java/org/gridsuite/loadflow/server/entities/LimitViolationEmbeddable.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@Getter
+@Embeddable
+public class LimitViolationEmbeddable {
+    @Column
+    private String subjectId;
+
+    @Column(name = "limit_")
+    private Double limit;
+
+    private String limitName;
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/loadflow/server/entities/LimitViolationEmbeddable.java:34]: added

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: change request for a suggestion
tests: ok

File: src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java [hunk 2]
```diff
@@ -173,6 +175,33 @@ private void cancelLoadFlowAsync(LoadFlowCancelContext cancelContext) {
...
+            .side(violation.getSide() != null ? violation.getSide().name() : "")
+            .limitType(violation.getLimitType()).build();
+    }
+
+    private List<LimitViolationInfos> getLimitViolations(Network network, LoadFlowRunContext loadFlowRunContext) {
+        List<LimitViolation> violations;
+        LoadFlowParameters lfCommonParams = buildParameters(loadFlowRunContext.getParameters(), loadFlowRunContext.getProvider());
+        Float limitReduction = loadFlowRunContext.getLimitReduction();
+        if (limitReduction == null) {
+            limitReduction = 0.001F;  // 0 is not allowed
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java]: Done

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowController.java [hunk 3]
```diff
@@ -53,6 +54,7 @@ public ResponseEntity<UUID> run(@Parameter(description = "Network UUID") @PathVa
                                     @Parameter(description = "Result receiver") @RequestParam(name = "receiver", required = false) String receiver,
                                     @Parameter(description = "reportUuid") @RequestParam(name = "reportUuid", required = false) UUID reportId,
                                     @Parameter(description = "reporterId") @RequestParam(name = "reporterId", required = false) String reportName,
+                                    @Parameter(description = "The limit reduction") @RequestParam(name = "limitReduction", required = false) Float limitReduction,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/loadflow/server/LoadFlowController.java]: Done

**Review by dbraquart — APPROVED** [review]
code: ok
tests: not done with last commit

**Pull request merged**